### DISCUSSION TITLE
How are people enabling HTTPS/SSL?

### DISCUSSION THREAD

**Discussion by rthorntn** [Part-DB/Part-DB-server#768] [Q&A] [answered]
Hi,

Apologies if this has already been covered somewhere.

I followed the Part-DB installation guide for Debian 11 (Bullseye) instructions after I failed to get the Docker container working when installing through Portainer (500 Error).

So my install is working, very cool, using http://192.168.1.1 but as I started to set it up I could see that an HTTPS setup is pretty much mandatory, DigiKey and the Scanner are the first two things I tried and neither work without HTTPS.  I went through the instructions and I didn't see anything talking about how to enable HTTPS?

This is a personal home setup on my NUC Proxmox server.  I'm happy in Linux, Docker not so much, so what are hobbyists doing to get part-db using https so digikey works?

Thanks!

**Comment by jbtronics** [comment, marked as answer]
If you have installed Part-DB directly, then you can follow any tutorial for setup TLS in apache (or nginx, depending). That is nothing special for Part-DB, thats bascially just normal webserver config:

Something like this: https://www.digitalocean.com/community/tutorials/how-to-secure-apache-with-let-s-encrypt-on-ubuntu

If you are using docker, you should put it behind a reverse proxy, which handels the HTTPs connection.

**Reply by gknit** [reply]
So I am using docker, and I tried ngrok and tunnelin. I can access my Part-DB instance over the internet via HTTPS, that works. However I still get the error:

`{"ErrorCode" : "invalid_request", "Error" :"Invalid redirection uri http://f28b-2a02-3100-422c-f600-15d4-a14-2c96-b995.ngrok-free.app/en/oauth/client/ip_digikey_oauth/check"}`

The redirection uri still keeps to be HTTP, despite the Part-DB URL being HTTPS with these reverse proxy apps. Can this be fixed somehow?

Thanks so much in advance, would appreciate it greatly if anyone could help.

Best,
Gero

**Comment by rthorntn** [comment]
Thanks, I followed this and have https now, the scanner works:

https://www.digitalocean.com/community/tutorials/how-to-create-a-self-signed-ssl-certificate-for-apache-in-ubuntu-16-04

I'm still wrestling with digikey, I get this error:

HTTP/1.1 401 Unauthorized returned for "https://api.digikey.com/Search/v3/Products/Keyword"

I have the digikey side setup, on there I made the Callback URL: https://10.10.1.21/ 
In .env.local I put the Client ID & Secret
When I click "Connect OAuth" I get "Success 11/19/24, 10:03 AM Connected to OAuth application successfully!"

Maybe the Digikey API need a internet routable/resolvable site with a properly signed SSL FQDN, absolutely no idea.

**Comment by rthorntn** [comment]
FIXED: Actually, at https://developer.digikey.com/ edit the Digikey app you created there and switch "Product Information V4" to None and "Product Information V3 (Deprecated)" to "Product Information" and save it, that worked for me.

**Comment by salfter** [comment]
I run my instance alongside a bunch of other services, with [caddy-docker-proxy](https://github.com/lucaslorentz/caddy-docker-proxy) to route traffic accordingly.  It also manages SSL.  The added configuration needed for PartDB amounts to just a few extra lines in docker-compose.yml:

```
    labels:
      caddy: partdb.alfter.us
      caddy.reverse_proxy: partdb.www:80
      caddy.log: partdb
      caddy.log.format: json
```

Everything works, including the DigiKey, LCSC, and Octopart integrations (haven't tested the others, but see no reason they wouldn't work).
